### ISSUE TITLE
Error when loading CRAM file with alignments to MT when reference NC_012920 is loaded

### ISSUE THREAD

**Issue by Schmytzi** [igvteam/igv#1573]
## Steps to reproduce
1. Select reference "NC_012920" in IGV
2. Load a CRAM file with mtDNA reads mapped to seqname "MT" ("chrM" and "chrMT" might be affected, too, not sure)

## Expected behavior
IGV should show the alignment and coverage tracks from that CRAM file

## Actual behavior
IGV stays blank and emits an error message (see below).

This issue does not arise with BAM files or other genomes. For instance, IGV properly detects that "MT" is equivalent to "chrM" in hg38.

## Error message
```
SEVERE [Sept 20,2024 10:26] [TrackLoader] java.lang.NullPointerException: Cannot read the array length because "bases" is null
        at org.igv/org.broad.igv.sam.cram.IGVReferenceSource.getReferenceBasesByRegion(IGVReferenceSource.java:66)
        at htsjdk@4.1.1/htsjdk.samtools.cram.build.CRAMReferenceRegion.fetchReferenceBasesByRegion(CRAMReferenceRegion.java:172)
        at htsjdk@4.1.1/htsjdk.samtools.cram.build.CRAMReferenceRegion.fetchReferenceBasesByRegion(CRAMReferenceRegion.java:195)
        at htsjdk@4.1.1/htsjdk.samtools.cram.structure.Slice.normalizeCRAMRecords(Slice.java:450)
        at htsjdk@4.1.1/htsjdk.samtools.cram.structure.Container.getSAMRecords(Container.java:322)
        at htsjdk@4.1.1/htsjdk.samtools.CRAMIterator.nextContainer(CRAMIterator.java:112)
        at htsjdk@4.1.1/htsjdk.samtools.CRAMIterator.hasNext(CRAMIterator.java:204)
        at htsjdk@4.1.1/htsjdk.samtools.SamReader$AssertingIterator.hasNext(SamReader.java:608)
        at org.igv/org.broad.igv.sam.reader.WrappedIterator.hasNext(WrappedIterator.java:49)
        at org.igv/org.broad.igv.sam.AlignmentUtils.firstAlignments(AlignmentUtils.java:243)
        at org.igv/org.broad.igv.sam.AlignmentDataManager.inferType(AlignmentDataManager.java:386)
        at org.igv/org.broad.igv.sam.AlignmentDataManager.<init>(AlignmentDataManager.java:91)
        at org.igv/org.broad.igv.track.TrackLoader.loadAlignmentsTrack(TrackLoader.java:879)
        at org.igv/org.broad.igv.track.TrackLoader.load(TrackLoader.java:179)
        at org.igv/org.broad.igv.ui.IGV.load(IGV.java:1235)
        at org.igv/org.broad.igv.ui.IGV.loadResources(IGV.java:1161)
        at org.igv/org.broad.igv.ui.IGV$2.run(IGV.java:357)
        at org.igv/org.broad.igv.util.LongRunningTask.submit(LongRunningTask.java:57)
        at org.igv/org.broad.igv.ui.IGV.loadTracks(IGV.java:368)
        at org.igv/org.broad.igv.batch.CommandExecutor.loadFiles(CommandExecutor.java:820)
        at org.igv/org.broad.igv.batch.CommandListener.processGet(CommandListener.java:427)
        at org.igv/org.broad.igv.batch.CommandListener.processClientSession(CommandListener.java:249)
        at org.igv/org.broad.igv.batch.CommandListener.run(CommandListener.java:140)
        at java.base/java.lang.Thread.run(Thread.java:1583)

SEVERE [Sept 20,2024 10:26] [IGV] Error loading track
SEVERE [Sept 20,2024 10:26] [IGV] org.broad.igv.exceptions.DataLoadException: Cannot read the array length because "bases" is null
        at org.igv/org.broad.igv.track.TrackLoader.load(TrackLoader.java:295)
        at org.igv/org.broad.igv.ui.IGV.load(IGV.java:1235)
        at org.igv/org.broad.igv.ui.IGV.loadResources(IGV.java:1161)
        at org.igv/org.broad.igv.ui.IGV$2.run(IGV.java:357)
        at org.igv/org.broad.igv.util.LongRunningTask.submit(LongRunningTask.java:57)
        at org.igv/org.broad.igv.ui.IGV.loadTracks(IGV.java:368)
        at org.igv/org.broad.igv.batch.CommandExecutor.loadFiles(CommandExecutor.java:820)
        at org.igv/org.broad.igv.batch.CommandListener.processGet(CommandListener.java:427)
        at org.igv/org.broad.igv.batch.CommandListener.processClientSession(CommandListener.java:249)
        at org.igv/org.broad.igv.batch.CommandListener.run(CommandListener.java:140)
        at java.base/java.lang.Thread.run(Thread.java:1583)
```

**Comment by jrobinso** [comment]
Have you tried loading "hg38" and switching to the MT "chromosome"?

**Comment by jrobinso** [comment]
Also, I can't reproduce this issue.   Could you verify you are using the lates version of IGV?   Is so, I will probably need a test case file.

**Comment by Schmytzi** [comment]
hg38 does not work but that is because the CRAM files were mapped to hg19. In this case, IGV correctly reports that the hashes of the references do not match. I realize that I might have implied otherwise when opening the issue. I apologize for that. 

hg19 works fine but our researchers insist on using NC_012920 because of its more comprehensive annotation. (The hash of hg19's MT chromosome also matches NC_012920 so that cannot be the underlying issue).

I am using version 2.18.2. The issue is present in version 2.16, too.

I have a test case (which is based on an individual from 1KGP IIRC) if you need one but it is too large to attach to this comment. Do you have a preferred way I should share it with you?

**Comment by jrobinso** [comment]
@Schmytzi if you send email to igv-team@broadinstitute.org I can send you a private dropbox link.  However I probably need only a single alignment in the CRAM file to reproduce the issue.  You could make such a CRAM with samtools.    However if you prefer to send the whole thing that is o.k., just send us an email.

**Comment by Schmytzi** [comment]
Thank you for saying that. I did not even think about only including a few reads. You can find a test case here:
[mito_subset.zip](https://github.com/user-attachments/files/17189346/mito_subset.zip)

The test case also includes a read pair on chr22. The reason is that I got different error messages depending on if the CRAM file includes multiple chromosomes or only MT.
The attached file will give you the aforementioned error but after subsetting for MT, the following error will appear in a dialog:
```
File <filename> does not contain any sequence names which match the current genome.
File:      1, 2, 3, 4, ...
Genome: NC_012920,
```
After closing the dialog, the tracks appear in IGV.

**Comment by jrobinso** [comment]
OK thanks.   I'm not sure I understand the comment on error messages.  If you load the cram with NC_012920 alignments only there are no sequence names in the CRAM file that match a sequence name in the reference.  If you include a chr22 alignment in the CRAM file and load hg38 there is at least 1 match, so you won't see that error message.

**Comment by jrobinso** [comment]
OK, the error is not what I thought it was.  The alias for MT <-> NC_012920 is actually recognized.   The error arises on initial load when IGV tries to sample alignments in the file to determine the experiment type (rna, 3rd gen, or dna).    It uses the sequence names in the CRAM header to find sequences to query alignments for.   The code for reading CRAM files does not handle the case of no alignments present on a listed sequence name well.    This is a bug obviously and we will fix it.

This also explains the confusing error message you are getting about "no matching sequence".

Here's the sequence names in the header in your file.   

```
@SQ	SN:1	LN:249250621	M5:1b22b98cdeb4a9304cb5d48026a85128
@SQ	SN:2	LN:243199373	M5:a0d9851da00400dec1098a9255ac712e
@SQ	SN:3	LN:198022430	M5:fdfd811849cc2fadebc929bb925902e5
@SQ	SN:4	LN:191154276	M5:23dccd106897542ad87d2765d28a19a1
@SQ	SN:5	LN:180915260	M5:0740173db9ffd264d728f32784845cd7
@SQ	SN:6	LN:171115067	M5:1d3a93a248d92a729ee764823acbbc6b
@SQ	SN:7	LN:159138663	M5:618366e953d6aaad97dbe4777c29375e
@SQ	SN:8	LN:146364022	M5:96f514a9929e410c6651697bded59aec
@SQ	SN:9	LN:141213431	M5:3e273117f15e0a400f01055d9f393768
@SQ	SN:10	LN:135534747	M5:988c28e000e84c26d552359af1ea2e1d
@SQ	SN:11	LN:135006516	M5:98c59049a2df285c76ffb1c6db8f8b96
@SQ	SN:12	LN:133851895	M5:51851ac0e1a115847ad36449b0015864
@SQ	SN:13	LN:115169878	M5:283f8d7892baa81b510a015719ca7b0b
@SQ	SN:14	LN:107349540	M5:98f3cae32b2a2e9524bc19813927542e
@SQ	SN:15	LN:102531392	M5:e5645a794a8238215b2cd77acb95a078
@SQ	SN:16	LN:90354753	M5:fc9b1a7b42b97a864f56b348b06095e6
@SQ	SN:17	LN:81195210	M5:351f64d4f4f9ddd45b35336ad97aa6de
@SQ	SN:18	LN:78077248	M5:b15d4b2d29dde9d3e4f93d1d0f2cbc9c
@SQ	SN:19	LN:59128983	M5:1aacd71f30db8e561810913e0b72636d
@SQ	SN:20	LN:63025520	M5:0dec9660ec1efaaf33281c0d5ea2560f
@SQ	SN:21	LN:48129895	M5:2979a6085bfe28e3ad6f552f361ed74d
@SQ	SN:22	LN:51304566	M5:a718acaa6135fdca8357d5bfe94211dd
@SQ	SN:X	LN:155270560	M5:7e0e2e580297b7764e31dbc80c2540dd
@SQ	SN:Y	LN:59373566	M5:1fa3474750af0948bdf97d5a0ee52e51
@SQ	SN:MT	LN:16569	M5:c68f52674c9fb33aef52dcf399755519
...
```

**Comment by Schmytzi** [comment]
Of course. Here it is: [mito_subset_bam.zip](https://github.com/user-attachments/files/17202034/mito_subset_bam.zip)

**Comment by jrobinso** [comment]
This should be fixed now with the latest release, 2.18.4.
